### PR TITLE
feat(admin-ui): refresh products when a `BatchJob` is completed

### PR DESCRIPTION
**Pull request by fPolic**

### CODE REVIEW

File: packages/admin-ui/ui/src/providers/import-refresh.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      const refreshedTimestamp = productListQuery.state.dataUpdatedAt
+
+      const completedJobs = batchJobs.filter(
+        (job) => job.status === "completed" && job.type === "product-import"
+      )
+
+      for (const job of completedJobs) {
+        const jobCompletedTimestamp = new Date(job.completed_at).getTime()
+        if (jobCompletedTimestamp > refreshedTimestamp) {
+          console.warn("Batch job finished: refreshing products list")
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/providers/import-refresh.tsx]: todo: rm
```suggestion
```

File: packages/admin-ui/ui/src/providers/import-refresh.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          queryClient.invalidateQueries(adminProductKeys.all)
```
Comment (olivermrbl) [inline packages/admin-ui/ui/src/providers/import-refresh.tsx:34]: question: Should we add the toast, if possible?

**Review by olivermrbl — APPROVED** [review]
Very nice. Only a single todo before we merge 👍

**Comment by olivermrbl** [comment]
I just noticed that the batch job is not updated in the drawer before opening it even though the products cache is successfully invalidated. 

Since the invalidation happens upon batch job completion, I would assume that the batch job in the drawer would similarly be in a completed state.

Attaching a screen recording below. Hope this makes sense :)

https://github.com/medusajs/medusa/assets/59018053/75c0ff25-bcb6-451b-b818-72d820d4759b

**Comment by fPolic** [comment]
> Since the invalidation happens upon batch job completion, I would assume that the batch job in the drawer would similarly be in a completed state.

Seems like stale `react-query` data, will look into it

File: packages/admin-ui/ui/src/providers/import-refresh.tsx (hunk 2 above)
Comment (fPolic) [reply packages/admin-ui/ui/src/providers/import-refresh.tsx:34]: done :)

**Comment by fPolic** [comment]
> > Since the invalidation happens upon batch job completion, I would assume that the batch job in the drawer would similarly be in a completed state.
> 
> Seems like stale `react-query` data, will look into it

Should be good now, the issue was with optimisations we did for polling of batch jobs

File: packages/admin-ui/ui/src/components/organisms/batch-jobs-activity-list/index.tsx [hunk 3]
```diff
@@ -68,6 +74,8 @@ const BatchJobActivityCard = (props: { batchJob: BatchJob }) => {
 
   const batchJob = useBatchJob(props.batchJob)
 
+  console.log(batchJob)
+
```
Comment (srindom) [inline packages/admin-ui/ui/src/components/organisms/batch-jobs-activity-list/index.tsx]: todo: rm

**Review by srindom — APPROVED** [review]
LGTM!

**Review by olivermrbl — APPROVED** [review]

**Pull request merged**